### PR TITLE
Remove containers/common/pkg/config from pkg/util

### PR DESCRIPTION
**Pull request by rhatdan**
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by rhatdan** [comment]
@mtrmac is this what you are thinking?

**Review by mtrmac — COMMENTED** [review]
> @mtrmac is this what you are thinking?

Something _vaguely_ along these lines.

AFAICS this doesn’t work yet because `…/podman/v5/util` is _also_ included by `vendor/github.com/containers/podman/v5/pkg/bindings/containers/create.go` , per

```go
package main

import (
	_ "github.com/containers/podman/v5/pkg/bindings"
	_ "github.com/containers/podman/v5/pkg/bindings/containers"
	_ "github.com/containers/podman/v5/pkg/bindings/generate"
	_ "github.com/containers/podman/v5/pkg/bindings/images"
	_ "github.com/containers/podman/v5/pkg/bindings/kube"
	_ "github.com/containers/podman/v5/pkg/bindings/manifests"
	_ "github.com/containers/podman/v5/pkg/bindings/network"
	_ "github.com/containers/podman/v5/pkg/bindings/play"
	_ "github.com/containers/podman/v5/pkg/bindings/pods"
	_ "github.com/containers/podman/v5/pkg/bindings/secrets"
	_ "github.com/containers/podman/v5/pkg/bindings/system"
	_ "github.com/containers/podman/v5/pkg/bindings/volumes"
)

func main() {

}
```

in a new directory + `go mod tidy && go mod vendor` and inspecting the result.

I didn’t check further how deep the rabbit hole goes; it might involve several more package splits. Or it might not.

---

We got here by having a jack-of-all-trades `pkg/util` package; I don’t think moving the code into a different `utils` package is _structurally_ an improvement. Move that into a separate `internal/config` (or maybe `pkg/config`? I think `internal` would be cleaner but it’s also not widely used).

---

That’s clearly not _this_ PR, but the Podman codebase has 29 other calls to `config.Default()`. If we have an `init` that exists on failure, it seems to me those 29 calls could be migrated to `DefaultContainerConfig` and we could remove some handling of unreachable errors.

… or maybe not? `config.Reload` exists… and if I am skimming the code correctly, that affects `config.Default()` but not `DefaultContainerConfig()`. I don’t understand what is going on. Maybe `DefaultContainerConfig` should not exist at all?

**Comment by Luap99** [comment]
If the goal is to completely get rid of c/common/pkg/config from pkg/bindings it is much more work
graph

generated with `goda graph "remote=1(reach(./pkg/bindings/...:all, github.com/containers/common/pkg/config))" | dot -Tsvg -o graph.svg` assuming this tool is correct but the result looks fine to me on quick look

File: cmd/podman/containers/container.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ var (
 		RunE:             validate.SubCommandExists,
 	}
 
-	containerConfig = util.DefaultContainerConfig()
```
Comment (Luap99) [inline cmd/podman/containers/container.go:23]: please use `registry.PodmanConfig().ContainersConfDefaultsRO` in all cmd/podman/... files instead

File: internal/config/config.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+)
+
+var containerConfig *config.Config
+
+func init() {
+	var err error
+	containerConfig, err = config.Default()
+	if err != nil {
+		logrus.Error(err)
+		os.Exit(1)
```
Comment (Luap99) [inline internal/config/config.go:17]: I really think this is just a miss feature abusing init(). Of course it is a pre existing issue but this doesn't feel right at all. For starters exit code 1 is wrong as we document 125 as default exit code. But in general storing a copy of the config is just wrong.

So please just convert all the callers to `config.Default()` and handle the errors there. As @mtrmac pointed this here does not work with config.Reload()

**Review by Luap99 — CHANGES_REQUESTED** [review]

**Review by Luap99 — APPROVED** [review]
LGTM

**Review by mtrmac — COMMENTED** [review]
ACK, this is clearly an improvement.

--- 

The test failure looks rather worrisome:
```
$ podman [options] stop --all -t 0
SIGABRT: abort
PC=0x40cd0e m=0 sigcode=0
```
with nothing obvious in the backtrace.

**Comment by Luap99** [comment]
Thank is a known flake: https://github.com/containers/podman/issues/23685

**Comment by baude** [comment]
/lgtm

thanks @rhatdan

**Pull request merged**